Sources/SwiftProtobuf/BinaryTypeAdditions.swift

```diff
@@ -19,7 +19,18 @@ import Foundation
 /// Messages
 ///
 public extension Message {
-    func serializedData() throws -> Data {
+    /// Serializes the message to the Protocol Buffer binary serialization format.
+    ///
+    /// - Parameters:
+    ///   - partial: The binary serialization format requires all `required` fields
+    ///     be present; when `partial` is `false`, `EncodingError.missingRequiredFields`
+    ///     is throw if any were missing. When `partial` is `true`, then partial
+    ///     messages are allowed, and `Message.isRequired` is not checked.
+    /// - Throws: An instance of `EncodingError` on failure .
+    func serializedData(partial: Bool = false) throws -> Data {
```

Comment:
A proto3 message could still have a proto2 message nested inside it, so `partial` needs to be supported there as well.